code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -394,7 +394,8 @@
 	taste_description = "ash"
 
 /datum/reagent/medicine/charcoal/on_mob_life(mob/living/M)
-	M.adjustToxLoss(-2*REM, 0)
+	if(prob(min(current_cycle-25, 10)))
```

Comment:
This doesn't heal less the longer it's in the mob. The probability increases until 35 cycles and then stays at 10%.